core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2373,6 +2376,13 @@ public UfsInfo getUfsInfo(long mountId) {
    * @param rpcContext the rpc context
    * @param inodePath the path for which metadata should be loaded
    * @param context the load metadata context
+   * @param statusCache UFS statuses which has already been retrieved for the {@code inodePath}
+   *                    This can be set to null if no status cache is available
+   * @param shouldParallelize if set to true and we are loading metadata for child inodes, then it
+   *                          will load metadata for separate paths concurrently. This should only
+   *                          be set to true if the calling thread doesn't already hold a write
+   *                          lock. If a write lock is held, and this parameter is true, this will
```

Comment:
`write lock is held`: Isn't there a way to see if a locked inode path has a write lock, and automatically be able to parallelize or not based on that result?

Reply:
As far as I'm aware, calling `#getLockPattern` gets the lock pattern for the currenty inode path, but doesn't tell us if there is a path further up in the tree that is write locked.
i.e. if you write-locked `/` but then load metadata on `/mnt` which is read locked, the `lockedInodePath#getLockPattern` will return `READ`. Deadlock can occur if any inode or edge in the current path is `WRITE_*`

Reply:
Yeah, I know. What if we added a method to LockedInodePath to see if there was any write lock along the path? That would help to automatically parallelize?

Reply:
I think that would be a welcome addition. It would probably simplify the current implementation